src/main/java/bio/terra/service/policy/TerraPolicyService.java

```diff
@@ -0,0 +1,114 @@
+package bio.terra.service.policy;
+
+import bio.terra.app.configuration.PolicyServiceConfiguration;
+import bio.terra.common.ExceptionUtils;
+import bio.terra.model.RepositoryStatusModelSystems;
+import bio.terra.policy.api.PublicApi;
+import bio.terra.policy.api.TpsApi;
+import bio.terra.policy.client.ApiException;
+import bio.terra.policy.model.TpsComponent;
+import bio.terra.policy.model.TpsObjectType;
+import bio.terra.policy.model.TpsPaoCreateRequest;
+import bio.terra.policy.model.TpsPolicyInputs;
+import bio.terra.service.policy.exception.PolicyConflictException;
+import bio.terra.service.policy.exception.PolicyServiceApiException;
+import bio.terra.service.policy.exception.PolicyServiceAuthorizationException;
+import bio.terra.service.policy.exception.PolicyServiceDuplicateException;
+import bio.terra.service.policy.exception.PolicyServiceNotFoundException;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TerraPolicyService {
+  private static final Logger logger = LoggerFactory.getLogger(TerraPolicyService.class);
+  private final PolicyServiceConfiguration policyServiceConfiguration;
+  private final PolicyApiService policyApiService;
+
+  public TerraPolicyService(
+      PolicyServiceConfiguration policyServiceConfiguration, PolicyApiService policyApiService) {
+    this.policyServiceConfiguration = policyServiceConfiguration;
+    this.policyApiService = policyApiService;
+  }
+
+  // -- Policy Attribute Object Interface --
+  public void createSnapshotPao(UUID snapshotId, @Nullable TpsPolicyInputs policyInputs) {
+    createPao(snapshotId, TpsObjectType.SNAPSHOT, policyInputs);
+  }
+
+  public void createPao(
+      UUID resourceId, TpsObjectType resourceType, @Nullable TpsPolicyInputs policyInputs) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsPolicyInputs inputs = (policyInputs == null) ? new TpsPolicyInputs() : policyInputs;
+
+    TpsApi tpsApi = policyApiService.getPolicyApi();
+    try {
+      tpsApi.createPao(
+          new TpsPaoCreateRequest()
+              .objectId(resourceId)
+              .component(TpsComponent.TDR)
+              .objectType(resourceType)
+              .attributes(inputs));
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  public void deletePao(UUID resourceId) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsApi tpsApi = policyApiService.getPolicyApi();
+    try {
+      try {
+        tpsApi.deletePao(resourceId);
+      } catch (ApiException e) {
+        throw convertApiException(e);
+      }
+    } catch (PolicyServiceNotFoundException e) {
+      // Not found should not cause a failure on delete.
+    }
```

Comment:
I would avoid nesting `try` when possible. Since the code is catching an exception that's throwing, you can check the exception type before throwing instead. I think this should work
```suggestion
    try {
      tpsApi.deletePao(resourceId);
    } catch (ApiException e) {
      RuntimeException exception = convertApiException(e);
      if (PolicyServiceNotFoundException.class.isAssignableFrom(exception.getClass())) {
        // Not found should not cause a failure on delete.
        return;
      }
      throw exception;
    }
```

This is still not great IMO since it's code that's trying to re-create some state which other code generated and then discarded. One approach to fix this is to add an argument to `convertApiException()` that tells it if it should ignore the duplicate case. At that point you could make `convertApiException()` throw the exception itself, or have it return `null` or an `Optional` so the caller can tell when nothing needs to be thrown.